src/transformers/models/lightglue/modeling_lightglue.py

```diff
@@ -0,0 +1,1158 @@
+        else:
+            matches, matching_scores = self._get_keypoint_matching(descriptors, mask, self.num_layers - 1)
+            final_pruned_keypoints_iterations = torch.ones_like(matching_scores) * self.num_layers
+
+        final_pruned_keypoints_iterations = final_pruned_keypoints_iterations.reshape(
+            batch_size, 2, initial_num_keypoints
+        )
+
+        return (
+            matches,
+            matching_scores,
+            final_pruned_keypoints_iterations,
+            all_hidden_states,
+            all_attentions,
+        )
+
+    @add_start_docstrings_to_model_forward(LIGHTGLUE_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        pixel_values: torch.FloatTensor = None,
+        labels: Optional[torch.LongTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, LightGlueKeypointMatchingOutput]:
+        loss = None
+        if labels is not None:
+            raise ValueError("LightGlue is not trainable, no labels should be provided.")
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if len(pixel_values.size()) != 5 or pixel_values.size(1) != 2:
+            raise ValueError("Input must be a 5D tensor of shape (batch_size, 2, num_channels, height, width)")
+
+        batch_size, _, channels, height, width = pixel_values.shape
+
+        list_keypoint_detection = [self.keypoint_detector(image_pair) for image_pair in pixel_values]
+
+        max_keypoints = max([keypoint_detection[0].shape[1] for keypoint_detection in list_keypoint_detection])
+        keypoints = torch.zeros(
+            (batch_size, 2, max_keypoints, 2), device=pixel_values.device, dtype=pixel_values.dtype
+        )
+        descriptors = torch.zeros(
+            (batch_size, 2, max_keypoints, self.config.descriptor_dim),
+            device=pixel_values.device,
+            dtype=pixel_values.dtype,
+        )
+        mask = torch.zeros((batch_size, 2, max_keypoints), device=pixel_values.device, dtype=torch.int)
+
+        for i, keypoint_detection_output in enumerate(list_keypoint_detection):
+            _keypoints, _scores, _descriptors, _mask = keypoint_detection_output[:4]
+            keypoints[i, :, : _keypoints.shape[1], :] = _keypoints
+            descriptors[i, :, : _descriptors.shape[1], :] = _descriptors
+            mask[i, :, : _mask.shape[1]] = _mask
+
+        absolute_keypoints = keypoints.clone()
+        absolute_keypoints[:, :, :, 0] = absolute_keypoints[:, :, :, 0] * width
+        absolute_keypoints[:, :, :, 1] = absolute_keypoints[:, :, :, 1] * height
+
```

Comment:
Can we do batch prediction as follows?

```python
pixel_values = pixel_values.reshape(-1, channels, height, width)
keypoints = self.keypoint_detector(pixel_values)
```
And then reshape keypoints/scores/etc

Reply:
I've changed the code to indeed use the keypoint_detector with batched input